linera-chain/src/certificate/mod.rs

```diff
@@ -38,93 +33,49 @@ pub type ConfirmedBlockCertificate = GenericCertificate<ConfirmedBlock>;
 /// Certificate for a [`Timeout`] instance.
 pub type TimeoutCertificate = GenericCertificate<Timeout>;
 
-/// A certified statement from the committee.
-pub type Certificate = GenericCertificate<CertificateValue>;
-
-impl Serialize for Certificate {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: serde::Serializer,
-    {
-        #[derive(Debug, Serialize)]
-        #[serde(rename = "Certificate")]
-        struct CertificateHelper<'a> {
-            value: &'a CertificateValue,
-            round: Round,
-            signatures: &'a Vec<(ValidatorName, Signature)>,
-        }
-
-        let helper = CertificateHelper {
-            value: self.inner(),
-            round: self.round,
-            signatures: self.signatures(),
-        };
-
-        helper.serialize(serializer)
-    }
-}
-
-impl<'de> Deserialize<'de> for Certificate {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        #[derive(Debug, Deserialize)]
-        #[serde(rename = "Certificate")]
-        struct CertificateHelper {
-            value: HashedCertificateValue,
-            round: Round,
-            signatures: Vec<(ValidatorName, Signature)>,
-        }
-
-        let helper: CertificateHelper = Deserialize::deserialize(deserializer)?;
-        if !is_strictly_ordered(&helper.signatures) {
-            Err(serde::de::Error::custom("Vector is not strictly sorted"))
-        } else {
-            Ok(Self::new(helper.value, helper.round, helper.signatures))
-        }
-    }
+/// Enum wrapping all types of certificates that can be created.
```

Comment:
We should still explain what a certificate is, too. Maybe add the old comment to the new one? In general, let's make sure the comments on the relevant types explain that:

* A certificate is a statement signed by a quorum of validators.
* A confirmed block certificate means that the block is finalized: It is the agreed block at that height on that chain.
* A validated block certificate means that the block is valid (but not necessarily finalized yet). Since they validate only one block per round, there can be at most one such certificate in every round.
* A timeout certificate means that the next consensus round has begun.

Reply:
Added more specific comments on the specific certificate type aliases rather than here.